hardware/oscilloscopes.py

```diff
@@ -20,7 +21,29 @@
 import visa
 from numpy import array, arange
 import time
+from hardware import u
 
+class MockOscilloscope:
+    def __init__(self, instr_name = None):
+        if not instr_name:
+            self.name = "Oscilloscope - Agilent DSO1024A "
+        else:
+            self.name = instr_name
+
+    def identify(self):
+        return self.name
+
+    def set_timeout(self, val):
+        self.timeout = val * u.milliseconds

```

Comment:
User should pass a value in milliseconds, but then the timeout function should convert to milliseconds, and pass just a `float` to `self.timeout`.

Double check whether you did this elsewhere.